kaldb/src/main/java/com/slack/kaldb/logstore/schema/SchemaAwareLogDocumentBuilderImpl.java

```diff
@@ -347,7 +359,137 @@ public static SchemaAwareLogDocumentBuilderImpl build(
+            : DEFAULT_LOG_MESSAGE_TYPE;
+
+    addField(doc, LogMessage.ReservedField.TYPE.fieldName, msgType, "", 0);
+
+    jsonMap.put(LogMessage.ReservedField.SERVICE_NAME.fieldName, indexName);
+    addField(doc, LogMessage.SystemField.INDEX.fieldName, indexName, "", 0);
+    addField(doc, LogMessage.ReservedField.SERVICE_NAME.fieldName, indexName, "", 0);
+
+    tags.remove(LogMessage.ReservedField.SERVICE_NAME.fieldName);
+    tags.remove(LogMessage.ReservedField.TYPE.fieldName);
+
+    for (Trace.KeyValue keyValue : tags.values()) {
+      if (keyValue.getVType() == Trace.ValueType.STRING) {
+        addField(doc, keyValue.getKey(), keyValue.getVStr(), "", 0);
+        jsonMap.put(keyValue.getKey(), keyValue.getVStr());
+      } else if (keyValue.getVType() == Trace.ValueType.BOOL) {
+        addField(doc, keyValue.getKey(), keyValue.getVBool(), "", 0);
+        jsonMap.put(keyValue.getKey(), keyValue.getVBool());
+      } else if (keyValue.getVType() == Trace.ValueType.INT32) {
+        addField(doc, keyValue.getKey(), keyValue.getVInt32(), "", 0);
+        jsonMap.put(keyValue.getKey(), keyValue.getVInt32());
+      } else if (keyValue.getVType() == Trace.ValueType.INT64) {
+        addField(doc, keyValue.getKey(), keyValue.getVInt64(), "", 0);
+        jsonMap.put(keyValue.getKey(), keyValue.getVInt64());
+      } else if (keyValue.getVType() == Trace.ValueType.FLOAT32) {
+        addField(doc, keyValue.getKey(), keyValue.getVFloat32(), "", 0);
+        jsonMap.put(keyValue.getKey(), keyValue.getVFloat32());
+      } else if (keyValue.getVType() == Trace.ValueType.FLOAT64) {
+        addField(doc, keyValue.getKey(), keyValue.getVFloat64(), "", 0);
+        jsonMap.put(keyValue.getKey(), keyValue.getVFloat64());
+      } else if (keyValue.getVType() == Trace.ValueType.BINARY) {
+        addField(doc, keyValue.getKey(), keyValue.getVBinary().toStringUtf8(), "", 0);
+        jsonMap.put(keyValue.getKey(), keyValue.getVBinary().toStringUtf8());
+      } else {
+        LOG.warn(
+            "Skipping field with unknown value type {} with key {}",
+            keyValue.getVType(),
+            keyValue.getKey());
+      }
+    }
+
+    LogWireMessage logWireMessage =
+        new LogWireMessage(indexName, msgType, message.getId().toStringUtf8(), timestamp, jsonMap);
+    final String msgString = JsonUtil.writeAsString(logWireMessage);
+    addField(doc, LogMessage.SystemField.SOURCE.fieldName, msgString, "", 0);
+    if (enableFullTextSearch) {
+      addField(doc, LogMessage.SystemField.ALL.fieldName, msgString, "", 0);
+    }
+
+    return doc;
+  }
+
+  // this should only have one reference from SpanFormatterTest
+  // We want to make sure none of our conversions changed subtly so keeping the old method alive for
+  // testing
+  // Once we get rid of LogMessage we can remove this and the associated test
+  @VisibleForTesting
+  public Document fromLogMessageTEST(LogMessage message) throws JsonProcessingException {
```

Comment:
It might be better to overload `fromMessage` with a `LogMesage` param, and then add a `@Deprecated` annotation to this method instead of suffixing this this `TEST`